Fix timed messages for busy actors

strohel opened the pull request:
### Add test showcasing delayed messages not delivered to busy actors 

This test fails (as of this commit) with
```
---- timed::tests::recurring_messages_for_busy_actors stdout ----
thread 'timed::tests::recurring_messages_for_busy_actors' panicked at 'assertion failed: `(left == right)`
  left: `[1, 3, 3]`,
 right: `[1, 2, 3, 2, 3]`', src/timed.rs:318:9
```

I.e. messages of type 2 (delayed) are never delivered.

### Process delayed/recurring messages also for busy actors

This fixes the recurring_messages_for_busy_actors test, and should fix #72.

---
_Please give the contained logic thorough review scrutiny, this is a central thing._

skywhale commented on src/timed.rs:
```diff
@@ -249,3 +260,73 @@ enum Payload<M> {
+                guard.push(message);
+            }
+
+            // Messages 1 or 3 are endless self-sending ones, keep to loop spinning.
```
typo?

skywhale reviewed (approved):
Thank you for finding the bug and addressing quickly. Code LGTM

strohel commented:
I've tested this in https://github.com/tonarino/portal/pull/2293 and it's looking good!

@skywhale (or anybody else) I've changed order of processing slightly, added/extended docs and fixed the typo, PTAL.

Once we merge this I'll do a release.

skywhale commented on src/timed.rs:
```diff
@@ -136,6 +136,12 @@ impl<M: Send + 'static, A: Actor<Context = TimedContext<M>, Message = M>> Timed<
 
             // Let inner actor do its job.
+            //
+            // Alternatively, we could send an `Instant` message to ourselves.
```
Thank you for additional documentation and agree with the current strategy.

skywhale commented on src/timed.rs:
```diff
@@ -161,24 +167,24 @@ impl<M: Send + 'static, A: Actor<Context = TimedContext<M>, Message = M>> Actor
 
-        self.queue.push(item);
+        // Process any expired items in the queue. In case that the actor is non-stop busy (there's
+        // always a message in its queue, perhaps because it sends a message itself in handle()),
```
a message "to" itself

strohel replied:
```suggestion
        // always a message in its queue, perhaps because it sends a message to itself in handle()),
```

skywhale reviewed (approved):
LGTM++

bschwind reviewed (approved):
LGTM!

As an aside, did you notice any change in the speed of the benchmarks?

strohel commented:
> As an aside, did you notice any change in the speed of the benchmarks?

We unfortunately don't have any benches that used the `Timed` wrapper (and this change is strictly contained to he timed wrapper).
